hal/modem_api.h

```diff
@@ -0,0 +1,88 @@
+/* mbed Microcontroller Library
```

Comment:
I dont understand this commit. The modem code looks to be very similar and only implemented for the C027 ublox modem. What if I bought https://www.u-blox.com/en/2g-and-3g-modules and hooked it upto a K64F? How do I make use of these drivers?

In general, CellularPPPModem should end up looking a lot like the ESP8266 I think assuming we're compatible with (and the modems are) the 3GPP 27.007 spec where a class in mbed OS which each modem mfgr can inherit from and customize for their particular modem. Those drivers may or may not live in mbed OS.